lib/improver/tests/nowcasting/lightning/test_NowcastLightning.py

```diff
@@ -0,0 +1,892 @@
+        expected.data[0, 8, 5:9] = [0.9, 0.9, 0.9, 1.]
+        return expected
+
+    def test_basic(self):
+        """Test that the method returns the expected cube type"""
+        result = self.plugin.process(CubeList([
+            self.fg_cube,
+            self.ltng_cube,
+            self.precip_cube]))
+        self.assertIsInstance(result, Cube)
+
+    def test_basic_with_vii(self):
+        """Test that the method returns the expected cube type when vii is
+        present"""
+        result = self.plugin.process(CubeList([
+            self.fg_cube,
+            self.ltng_cube,
+            self.precip_cube,
+            self.vii_cube]))
+        self.assertIsInstance(result, Cube)
+
+    def test_no_first_guess_cube(self):
+        """Test that the method raises an error if the first_guess cube is
+        omitted from the cubelist"""
+        msg = (r"Got 0 cubes for constraint Constraint\(name=\'probability_of_"
+               r"lightning\'\), expecting 1.")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            self.plugin.process(CubeList([
+                self.ltng_cube,
+                self.precip_cube]))
+
+    def test_no_lightning_cube(self):
+        """Test that the method raises an error if the lightning cube is
+        omitted from the cubelist"""
+        msg = (r"Got 0 cubes for constraint Constraint\(name=\'rate_of_"
+               r"lightning\'\), expecting 1.")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            self.plugin.process(CubeList([
+                self.fg_cube,
+                self.precip_cube]))
+
+    def test_no_precip_cube(self):
+        """Test that the method raises an error if the precip cube is
+        omitted from the cubelist"""
+        msg = (r"Got 0 cubes for constraint Constraint\(name=\'probability_of_"
+               r"precipitation\'\), expecting 1.")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            self.plugin.process(CubeList([
+                self.fg_cube,
+                self.ltng_cube]))
+
+    def test_precip_has_no_thresholds(self):
+        """Test that the method raises an error if the threshold coord is
+        omitted from the precip_cube"""
+        self.precip_cube.remove_coord('threshold')
+        msg = (r"Cannot find prob\(precip > 0.5\) cube in cubelist.")
+        with self.assertRaisesRegex(ConstraintMismatchError, msg):
+            self.plugin.process(CubeList([
+                self.fg_cube,
+                self.ltng_cube,
+                self.precip_cube]))
+
+    def test_result_with_vii(self):
+        """Test that the method returns the expected data when vii is
+        present"""
+        # Set precip_cube forecast period to be zero.
+        self.precip_cube.coord('forecast_period').points = [0.]
+        expected = self.set_up_vii_input_output()
+
+        # No halo - we're only testing this method.
+        plugin = Plugin(2000.)
```

Comment:
There is a halo being set here.

Reply:
This is a terminology problem. The radius is equivalent to one grid-cell, so exactly one grid-cell is used in the neighbourhooding. This means that there is no halo around the cell. I've updated the comment to make this a little clearer.